esp/services/ws_smc/ws_smcService.cpp

```diff
@@ -2344,3 +2345,175 @@ void CWsSMCEx::setActiveWUs(IEspContext &context, IEspActiveWorkunit& wu, IEspAc
     }
 }
 
+static const char *LockModeNames[] = { "READ", "WRITE", "HOLD", "SUB" };
+
+void CWsSMCEx::addLockInfo(CLockMetaData& lD, const char* xPath, unsigned msNow, time_t ttNow, IArrayOf<IEspLock>& locks)
+{
+    Owned<IEspLock> lock = createLock();
+    lock->setXPath(xPath);
```

Comment:
This is a good point. I am going to add 'LogicalFile' to the 'Lock' structure. For LFN, I use setLogicalFile(). For non-LFN, I use setXPath().